### PR TITLE
Add default parameter for get_group

### LINKED ISSUE #9299: Exception in get_group() with invalid index
Hello,

I'm using the pandas function "pandas.core.groupby.GroupBy.get_group()" to parallelize my code.
This function raises a "KeyError" exception if the passed index is not contained in the index list of the grouped DataFrame.

A basic example is:
data = DataFrame({'ind':[0,0,2,2]})
gb=a.groupby('ind')
gb.get_group(1)

Which raises "KeyError: 1"

In order to use intensively the "get_group()" function in code parallelization, an option to avoid this exception would be very useful, similar to "error_bad_lines=False" of the "read_csv()" function.

An empty DataFrame could be returned by the "get_group()" function when this option and an invalid index are given.

### PR DESCRIPTION
**Pull request by kinow**
- [x] closes #9299
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by kinow** [comment]
This is for #9299, but there was a previous attempt in #5452 where it was suggested in the final comments that perhaps it should be similar to how `#get()` works.

My initial attempt simply returns the `default` when not `None`. The `#get()`actually catches exceptions and returns the default instead.

https://github.com/pandas-dev/pandas/blob/12859382a4a5325e6fd19a209b128bab44a3a37a/pandas/core/generic.py#L3818-L3821

Let me know if you think that would be a better approach, and I will update it :+1: 

Thanks!
Bruno

File: pandas/core/groupby/groupby.py [hunk 1]
```diff
@@ -671,6 +671,9 @@ def get_group(self, name, obj=None):
             the DataFrame to take the DataFrame out of.  If
             it is None, the object groupby was called on will
             be used
+        default : object
+            the default object in case the group requested is
+            not found
```
Comment (TomAugspurger) [inline pandas/core/groupby/groupby.py]: Probably needs to end in a period.

File: pandas/core/groupby/groupby.py [hunk 2]
```diff
@@ -671,6 +671,9 @@ def get_group(self, name, obj=None):
             the DataFrame to take the DataFrame out of.  If
             it is None, the object groupby was called on will
             be used
+        default : object
+            the default object in case the group requested is
```
Comment (TomAugspurger) [inline pandas/core/groupby/groupby.py]: "the group requested" -> \`name\`

**Review by TomAugspurger — COMMENTED** [review]
Looks good. A couple doc comments.

Needs a release note in `1.0.0.rst` under "Other enhancements".

File: pandas/core/groupby/groupby.py (hunk 1 above)
Comment (WillAyd) [reply pandas/core/groupby/groupby.py]: Should this be a more specific type? Would there ever be a reason to pass something else besides a `get_group` callable as the default?

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/core/groupby/groupby.py (hunk 2 above)
Comment (kinow) [reply pandas/core/groupby/groupby.py]: Fixed, thanks!

File: pandas/core/groupby/groupby.py (hunk 1 above)
Comment (kinow) [reply pandas/core/groupby/groupby.py]: >Probably needs to end in a period.

Fixed, thanks!

Comment (kinow) [reply pandas/core/groupby/groupby.py]: >Should this be a more specific type? Would there ever be a reason to pass something else besides a get_group callable as the default?

I think you could also return a fixed value, like `default=DataFrame([-1, 0])`?

**Comment by WillAyd** [comment]
Thinking through this the code itself is good I guess I'm just wondering why this is generally useful? Is there a real use case for this that you can think of?

**Comment by kinow** [comment]
>Thinking through this the code itself is good I guess I'm just wondering why this is generally useful? Is there a real use case for this that you can think of?

This issue never affected me. The use case for this appears to be related to code parallelization, reported in #9299

**Comment by WillAyd** [comment]
Read through the history of that and I guess same thing where it's unclear how this is practical. I see the consistency with `.get` being a motivating factor but I don't understand why returning another group would be more useful than returning `None`

**Comment by WillAyd** [comment]
Thanks for the PR but I don't think this is really useful, so closing. Ping if you disagree and can pick back up

**Comment by kinow** [comment]
No problems @WillAyd !  Cheers :+1:

**Pull request closed without merging**